Lib/xml/etree/ElementTree.py

```diff
@@ -1726,6 +1731,18 @@ def close(self):
             del self.parser, self._parser
             del self.target, self._target
 
+    def flush(self):
+        was_enabled = self.parser.GetReparseDeferralEnabled()
+
+        if was_enabled:
+            self.parser.SetReparseDeferralEnabled(False)
+        try:
+            self.parser.Parse(b"", False)
+        except self._error as v:
+            self._raiseerror(v)
+        finally:
+            if was_enabled:
+                self.parser.SetReparseDeferralEnabled(True)
```

Comment:
I do not know whether you wrote this in such form intentionally, but it can be written without `if`s:
```python
        was_enabled = self.parser.GetReparseDeferralEnabled()
        try:
            self.parser.SetReparseDeferralEnabled(False)
            self.parser.Parse(b"", False)
        except self._error as v:
            self._raiseerror(v)
        finally:
            self.parser.SetReparseDeferralEnabled(was_enabled)
```

Reply:
@serhiy-storchaka it was intentional, yes:
- The suggestion has beauty to it but it arguably requires more thinking that the original, not less.
- The suggestion puts more things into try-except scope while there are no exceptions to catch (and so this is misleading and only okay because of inside knowledge that other readers do not have) so it should be outside in my view, from a purity point of view.
- I wanted to avoid needless function calls when there is nothing to restore. Your approach relies on idempotency and the call being fast. Is probably the case, but is whitebox knowledge while the test was blackbox before.

How would you like to proceed? Still the exact same proposal?

PS: I'm adding and dropping `PY_STDLIB_MOD_SET_NA([_elementtree])` here to build with and without the C module. Can I be sure the CI covers both? And any ideas how to be sure which one I have at runtime?

Reply:
> > Can I be sure the CI covers both?

@serhiy-storchaka PS: That I can answer with "yes" no, CI covers both as proven by the C-related failure. Fix pushed.

Reply:
> I do not know whether you wrote this in such form intentionally, but it can be written without `if`s:
> 
> ```python
>         was_enabled = self.parser.GetReparseDeferralEnabled()
>         try:
>             self.parser.SetReparseDeferralEnabled(False)
>             self.parser.Parse(b"", False)
>         except self._error as v:
>             self._raiseerror(v)
>         finally:
>             self.parser.SetReparseDeferralEnabled(was_enabled)
> ```

@serhiy-storchaka PPS: Applied x2 now. Maybe you're right about readability and it's probably fast enough. If you change your mind, I can adjust it once more.

Reply:
I was just asking in case you overlooked this option, which I think is simpler.

There is a value in placing `SetReparseDeferralEnabled()` inside the `try` block. If there is an interruption (user pressed Ctrl-C) between `SetReparseDeferralEnabled()` and `Parse()`, there is larger chance to restore the initial value. Of course, nothing will help if the `finally` block interrupted, but this is a different hard issue.

All ElemetTree tests are rub with and without the C implementation. There are two files: `test_xml_etree.py` and `test_xml_etree_c.py`. The former imports ElementTree with the C implementation blocked (see `setUpModule`). The latter runs the same tests with the C implementation unblocked. You can use `ET is pyET` to test that the current implementation is pure Python.

Reply:
@serhiy-storchaka thanks for elaborating, replaced by `ET is pyET` now.